### PR TITLE
add code-generaion evaluation for woq gptq

### PR DESCRIPTION
**Pull request by changwangss**
## Type of Change

feature or bug fix or documentation or validation or others  
API changed or not

## Description

detail description

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

**Review by hshen14 — APPROVED** [review]

File: examples/pytorch/nlp/huggingface_models/language-modeling/quantization/llm/run_clm_no_trainer.py [hunk 1]
```diff
@@ -78,7 +78,40 @@
                                                                            this should align with your model config, \
                                                                            and your dataset builder args: args.pad_max_length')
 parser.add_argument('--gptq_debug', action='store_true', help='Whether to use debug model ')
-# =======================================
+# ==============code generation args===========
+parser.add_argument("--code_generation", action="store_false")
+parser.add_argument("--n_samples", default=200, type=int)
```
Comment (chensuyue) [inline examples/pytorch/nlp/huggingface_models/language-modeling/quantization/llm/run_clm_no_trainer.py:83]: Can we just add parameters that the script required?

**Comment by chensuyue** [comment]
@YIYANGCAI as we synced you will refine this gptq_debug path, can you review this PR?

File: examples/pytorch/nlp/huggingface_models/language-modeling/quantization/llm/run_clm_no_trainer.py (hunk 1 above)
Comment (changwangss) [reply examples/pytorch/nlp/huggingface_models/language-modeling/quantization/llm/run_clm_no_trainer.py:83]: it is used with line 322, the parameters requested by bigcode-evaluation repo.
https://github.com/bigcode-project/bigcode-evaluation-harness

**Comment by ftian1** [comment]
I am little confused why such example is in INC but not ITREX?

**Comment by changwangss** [comment]
> I am little confused why such example is in INC but not ITREX?

itrex gptq is not ready now. so use inc first, this PR is not ready to merge and @YIYANGCAI will refine this example, I think he will cover this part (evaluation for code-generation).

**Review by YIYANGCAI — APPROVED** [review]

**Comment by chensuyue** [comment]
@changwangss please fix the bug for tasks input.

**Review by chensuyue — APPROVED** [review]

**Pull request merged**